.github/workflows/vuln-check-reusable.yaml

```diff
@@ -0,0 +1,164 @@
+          if [[ ${{ inputs.find-latest-release }} = 'true' ]]; then
+            releases=$($SCRIPT_PATH/fetch_gh_releases "${{ github.event.repository.owner.login }}" "${{ github.event.repository.name }}")
+            echo -------------
+            echo "releases: $releases"
+            echo -------------
+            target_release=''
+            if [[ -n $releases ]]; then
+              target_release=$($SCRIPT_PATH/find_latest_release "${{ inputs.target-ref }}" $releases)
+            fi
+            if [[ -z $target_release ]]; then
+              echo "Can't find a target release"
+              exit 1
+            fi
+            echo $target_release
+            echo "target-ref=$target_release" >> $GITHUB_OUTPUT
+          else
+            echo "target-ref=${{ inputs.target-ref }}" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Checkout the target repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ steps.prepare-target-ref.outputs.target-ref }}
+
+      - name: Set up JDK 8
+        uses: actions/setup-java@v4
+        with:
+          java-version: '8'
+          distribution: 'temurin'
+
+      - name: Docker build
+        uses: gradle/gradle-build-action@v3
+        with:
+          arguments: docker
+
+      - name: Set version
+        id: version
+        run: |
+          VERSION=$(${{ inputs.version-command }})
+          echo "version=${VERSION}" >> $GITHUB_OUTPUT
+
+  vuln-check:
+    runs-on: ubuntu-latest
+
+    needs: build
+
+    strategy:
+      matrix:
+        image: ${{ fromJSON(inputs.images) }}
+      fail-fast: false
+      max-parallel: 1
+
+    continue-on-error: true
+
+    steps:
+      - name: Check if Slack webhook URL is set
+        id: slack-webhook
+        run: |
+          if [[ -n "${{ secrets.SLACK_SECURITY_WEBHOOK_URL }}" ]]; then
+            echo "Slack webhook URL is set. Post a message if the vulnerability check fails."
+            echo "enabled=true" >> $GITHUB_OUTPUT
+          else
+            echo "Slack webhook URL is not set. Skip posting a message."
+            echo "enabled=false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.CR_PAT }}
+
+      - name: Run Trivy vulnerability scanner for ${{ matrix.image[0] }}
+        uses: aquasecurity/trivy-action@master
+        with:
+          image-ref: ghcr.io/scalar-labs/${{ matrix.image[1] }}:${{ needs.build.outputs.version }}
+          format: 'table'
+          exit-code: '1'
+          ignore-unfixed: true
+          severity: 'CRITICAL,HIGH'
+          timeout: '60m'
+
+      - name: Post Trivy vulnerability check failure for ${{ matrix.image[0] }}
+        if: failure() && steps.slack-webhook.outputs.enabled == 'true'
```

Comment:
Added a step to check if the webhook URL is specified since [it is not allowed to access the `secrets` from the `if:` condition direcly.](https://docs.github.com/en/actions/security-for-github-actions/security-guides/using-secrets-in-github-actions#using-secrets-in-a-workflow)